src/engine/guidance/post_processing.cpp

```diff
@@ -60,7 +60,8 @@ inline bool choiceless(const RouteStep &step, const RouteStep &previous)
                                                    step.intersections.front().entry.end(),
                                                    true);
 
-    return is_without_choice;
+    return is_without_choice && step.maneuver.instruction.type != TurnType::EndOfRoad;
+    ;
```

Comment:
No, on roads without a name, the `isLinkroad` can trigger, even though it is a longer road. The image linked above will still not show the turn, even though it might seem just the same, without the additional modifications. The two tests added reflect these two cases.